test/test-server/src/index.ts

```diff
@@ -0,0 +1,401 @@
+                    ...config?.emailDelivery,
+                    service: {
+                        ...config?.emailDelivery?.service,
+                        sendEmail: eval(`${config?.emailDelivery?.service?.sendEmail}`),
+                    },
+                };
+            }
+            recipeList.push(Passwordless.init(init));
+        }
+        if (recipe.recipeId === "multifactorauth") {
+            recipeList.push(MultiFactorAuth.init(config));
+        }
+        if (recipe.recipeId === "totp") {
+            recipeList.push(TOTP.init(config));
+        }
+    });
+
+    settings.recipeList = recipeList;
+
+    if (settings.supertokens?.networkInterceptor) {
+        settings.supertokens.networkInterceptor = eval(`${settings.supertokens.networkInterceptor}`);
+    }
+
+    supertokens.init(settings);
+}
+
+supertokens.init({
+    ...defaultConfig,
+    supertokens: {
+        connectionURI: process.env.ST_CONNECTION_URI || "http://localhost:8080",
+    },
+    recipeList: [Session.init()],
+});
+
+const app = express();
+app.use(express.json());
+app.use((req, res, next) => {
+    logDebugMessage(req.method, req.path);
+    next();
+});
+app.use(middleware());
+app.use(errorHandler());
+
+app.get("/test/ping", async (req, res, next) => {
+    res.json({ ok: true });
+});
+
+app.post("/test/init", async (req, res, next) => {
+    initST(req.body.config);
+    res.json({ ok: true });
+});
+
+app.get("/test/overrideparams", async (req, res, next) => {
+    let sessionVars = getSessionVars();
+    const overrideparams: OverrideParamsType = {
+        sendEmailToUserId,
+        token,
+        userPostPasswordReset,
+        emailPostPasswordReset,
+        sendEmailCallbackCalled,
+        sendEmailToUserEmail,
+        sendEmailToRecipeUserId,
+        userInCallback,
+        email,
+        newAccountInfoInCallback,
+        primaryUserInCallback: primaryUserInCallback?.toJson(),
+        userIdInCallback: userIdInCallback ?? sessionVars.userIdInCallback,
+        recipeUserIdInCallback:
+            recipeUserIdInCallback?.getAsString() ??
+            recipeUserIdInCallback ??
+            sessionVars.recipeUserIdInCallback?.getAsString(),
+        info,
+        store,
+    };
+    res.json(overrideparams);
+});
+
+app.post("/test/resetoverrideparams", async (req, res, next) => {
+    resetOverrideParams();
+    res.json({ ok: true });
+});
+
+app.post("/test/mockexternalapi", async (req, res, next) => {
+    const { url, status, body, path, method } = req.body;
+    nock(url)[method](path).reply(status, body);
+    res.json({ ok: true });
+});
+
+app.get("/test/waitforevent", async (req, res, next) => {
+    try {
+        logDebugMessage("ProcessState:waitForEvent %j", req.query);
+        const instance = ProcessState.getInstance();
+        const eventEnum = req.query.event ? Number(req.query.event) : null;
+        const event = eventEnum ? await instance.waitForEvent(eventEnum) : undefined;
```

Comment:
We should make the event param required and throw in that case instead of skipping the entire function and returning immediately. It signifies a test coding error and it could make test pass when they shouldn't.

Reply:
make sense, done.